src/main/groovy/be/cytomine/software/communication/SSH.groovy

```diff
@@ -44,22 +55,28 @@ class SSH implements Communication {
             }
             Properties properties = new Properties()
             properties.put("StrictHostKeyChecking", "no")
-            Session session = jSch.getSession(user, host, port)
+            session = jSch.getSession(user, host, port)
 
             session.setConfig(properties)
-            log.info("connection before with: $user@$host:$port")
+            log.info("Connection with: $user@$host:$port")
             session.connect()
-            log.info("connection to the session O")
+            log.info("Connection ok!")
             return session
         } catch (JSchException ex) {
             log.info(ex.toString())
             return null
         }
     }
 
+    @Override
+    def getExitStatus()
+    {
+        return statusCode
+    }
+
     @Override
     def executeCommand(String command) throws JSchException, UnknownHostException {
-        Session session = createSession()
+        session = createSession()
```

Comment:
So you create session in constructor and re-create it here. 
Please make something efficient. Static method ? Same for other methods...